src/cpp/rtps/security/SecurityManager.cpp

```diff
@@ -736,9 +747,11 @@ void SecurityManager::remove_participant(
             authentication_plugin_->return_sharedsecret_handle(shared_secret_handle, exception);
         }
 
-        remove_discovered_participant_info(dp_it->second.get_auth());
+        DiscoveredParticipantInfo::AuthUniquePtr remote_participant_info = dp_it->second.get_auth();
```

Comment:
```suggestion
        // AuthUniquePtr must be removed after unlock SecurityManager's mutex.
        // This is to avoid a deadlock with TimedEvents.
        DiscoveredParticipantInfo::AuthUniquePtr remote_participant_info = dp_it->second.get_auth();
```